security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -236,7 +236,7 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 		// If working as Citadel agent, send request for normal key/cert pair.
 		// If working as ingress gateway agent, fetch key/cert or root cert from SecretFetcher. Resource name for
 		// root cert ends with "-cacert".
-		ns, err := sc.generateSecret(ctx, token, connKey, time.Now())
+		ns, err := sc.generateSecret(ctx, token, connKey, time.Now(), true)
```

Comment:
Where is this withToken set as false?

Reply:
https://github.com/istio/istio/blob/4d2d350a9239ad4897122a53920b5a5032fcf66a/security/pkg/nodeagent/cache/secretcache.go#L594

when user enables the cert rotate feature